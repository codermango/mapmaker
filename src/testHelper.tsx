import { render } from "@testing-library/react";
import { ReactElement } from "react";
import { DndProvider } from "react-dnd";
import { HTML5Backend } from "react-dnd-html5-backend";
import { Provider } from "react-redux";
import { store } from "./app/store";

export const renderWithRedux = (component: ReactElement) => {
  return render(<Provider store={store}>{component}</Provider>);
};

export const renderWithDnd = (component: ReactElement) => {
  return render(<DndProvider backend={HTML5Backend}>{component}</DndProvider>);
};

export const renderWithReduxAndDnd = (component: ReactElement) => {
  return render(
    <Provider store={store}>
      <DndProvider backend={HTML5Backend}>{component}</DndProvider>
    </Provider>
  );
};
