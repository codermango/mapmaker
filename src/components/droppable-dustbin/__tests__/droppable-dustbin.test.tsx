import { renderWithDnd } from "../../../testHelper";
import DroppableDustbin from "../droppable-dustbin";

test("render correctly", () => {
  expect(
    renderWithDnd(<DroppableDustbin onDeleteAsset={jest.fn()} />).asFragment()
  ).toMatchSnapshot();
});
