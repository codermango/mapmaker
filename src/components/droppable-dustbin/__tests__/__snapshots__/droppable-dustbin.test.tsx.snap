// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<DocumentFragment>
  <img
    alt="dustbin"
    class="deleteImg"
    src="delete.png"
  />
</DocumentFragment>
`;
